src/jit/flowgraph.cpp

```diff
@@ -5516,6 +5586,14 @@ unsigned Compiler::fgMakeBasicBlocks(const BYTE* codeAddr, IL_OFFSET codeSize, F
 #endif // !FEATURE_CORECLR && _TARGET_AMD64_
                     }
 
+                    if (fgCanSwitchTier0ToTier1() && fgMayExplicitTailCall())
```

Comment:
Was there a specific case where this was observed?

Reply:
The new test has an explicit tail call and tries to overflow the stack if a tail call is not generated, and for the test QuickJit and QuickJitForLoops are enabled. It failed on Linux arm with stack overflow. I didn't see an issue on x64, where a tier 0 compilation generated a tail call (https://github.com/dotnet/coreclr/pull/23597#issuecomment-481416442 has more info).

Reply:
Great - thanks!